specs/~dynamic-names.yml

```
@@ -0,0 +1,349 @@
+overview: |
+  Rationale: this special notation was introduced primarily to allow the dynamic
+  loading of partials. The main advantage that this notation offers is to allow
+  dynamic loading of partials, which is particularly useful in cases where
+  polymorphic data needs to be rendered in different ways, cases which would
+  otherwise be possible to render only with solutions that are convoluted,
+  inefficient, or both.
+  Example.
+  Let's consider the following data:
+    items: [
+      { content: 'Hello, World!' },
+      { url: 'http://example.com/foo.jpg' },
+      { content: 'Some text' },
+      { content: 'Some other text' },
+      { url: 'http://example.com/bar.jpg' },
+      { url: 'http://example.com/baz.jpg' },
+      { content: 'Last text here' }
+    ]
```

I think code blocks need to be indented by at least four spaces in order for most Markdown implementations to recognize them as such.

```suggestion
      items: [
        { content: 'Hello, World!' },
        { url: 'http://example.com/foo.jpg' },
        { content: 'Some text' },
        { content: 'Some other text' },
        { url: 'http://example.com/bar.jpg' },
        { url: 'http://example.com/baz.jpg' },
        { content: 'Last text here' }
      ]
```